src/pykeen/datasets/ea/combination.py

```diff
@@ -0,0 +1,176 @@
+
+        # extra alignment relation
+        relation_to_id["same-as"] = relation_offset
+        # filter alignment
+        mask = ~(alignment[EA_SIDE_LEFT].isin(left.entity_to_id) & alignment[EA_SIDE_RIGHT].isin(right.entity_to_id))
+        if mask.any():
+            alignment = alignment.loc[~mask]
+            logger.warning(
+                f"Dropped {format_relative_comparison(part=mask.sum(), total=alignment.shape[0])} "
+                f"alignments due to unknown labels.",
+            )
+        # map alignment to (new) IDs
+        left_id = (
+            alignment[EA_SIDE_LEFT].apply(left.entity_to_id.__getitem__) + entity_offsets[0]
+        )  # offset should be zero
+        right_id = alignment[EA_SIDE_RIGHT].apply(right.entity_to_id.__getitem__) + entity_offsets[1]
+        # append alignment triples
+        mapped_triples.append(
+            torch.stack(
+                [
+                    torch.as_tensor(data=left_id.values, dtype=torch.long),
+                    torch.full(size=(len(left_id),), fill_value=relation_offset),
+                    torch.as_tensor(data=right_id.values, dtype=torch.long),
+                ],
+                dim=-1,
+            )
+        )
+        # merged factory
+        return TriplesFactory(
+            mapped_triples=torch.cat(mapped_triples, dim=0),
+            entity_to_id=entity_to_id,
+            relation_to_id=relation_to_id,
+            **kwargs,
+        ), torch.stack([left_id, right_id])
+
+
+class CollapseGraphPairCombinator(GraphPairCombinator):
+    """This combinator merges all matching entity pairs into a single ID."""
+
+    def __call__(
+        self,
+        left: TriplesFactory,
+        right: TriplesFactory,
+        alignment: pandas.DataFrame,
+        **kwargs,
+    ) -> TriplesFactory:  # noqa: D102
+        # concatenate (shifted) triples
+        mapped_triples = torch.cat(
+            [
+                left.mapped_triples,
+                right.mapped_triples
+                + torch.as_tensor([left.num_entities, left.num_relations, left.num_entities]).view(1, 3),
+            ]
+        )
+        # determine connected components regarding the same-as relation (i.e., applies transitivity)
+        id_mapping = torch.arange(left.num_entities + right.num_entities)
+        for cc in get_connected_components(
+            pairs=(
+                (
+                    left.entity_labeling.label_to_id[row[EA_SIDE_LEFT]],
+                    right.entity_labeling.label_to_id[row[EA_SIDE_RIGHT]] + left.num_entities,
+                )
+                for _, row in alignment.iterrows()
+            )
+        ):
+            cc = list(cc)
+            id_mapping[cc] = min(cc)
+        # apply id mapping
+        h, r, t = mapped_triples.t()
+        h, t = id_mapping[h], id_mapping[t]
+        # ensure consecutive IDs
+        unique, inverse = torch.cat([h, t]).unique(return_inverse=True)
+        h, t = inverse.split(split_size_or_sections=2)
```

Comment:
The keyword argument for `split` when using a tensor is apparently `split_size` (see [here](https://pytorch.org/docs/stable/generated/torch.Tensor.split.html?highlight=split#torch.Tensor.split))